### PR TITLE
feat: debounce author profile paging

### PR DESCRIPTION
**Pull request by BernieSumption**
1. Debounce author profile paging within the UI component - the display page (Showing 1 to 20 of 41) updates immediately, the query is delayed until no page changes have hapened in the last 250ms
2. Refactor pagination component to use onChangePage instead of onNext and onPrev
3. Fixed some errors relating to article publication date formats that had found their way into master after PR 525

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/63e9c7f4dc863450

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/fe70cbfb0feae19e

File: packages/pagination/__tests__/pagination.web.test.js [hunk 1]
```diff
@@ -1,5 +1,60 @@
 import "jest-styled-components";
+import React from "react";
+import Enzyme, { mount } from "enzyme";
+import React16Adapter from "enzyme-adapter-react-16";
```
Comment (colincclark) [inline packages/pagination/__tests__/pagination.web.test.js]: Utilisation of jest-configurator and a `jest.config.js` would negate the need for this boilerplate setup

File: packages/pagination/pagination.js [hunk 2]
```diff
@@ -40,67 +40,98 @@ const styles = StyleSheet.create({
...
-          styles.border,
-          !hideTopKeyline && styles.borderTop,
-          !hideBottomKeyline && styles.borderBottom
-        ]}
-      >
-        <View>{prevComponent}</View>
-        <View>{nextComponent}</View>
+const debounceIntervalMs = 250;
+
+class Pagination extends React.Component {
```
Comment (colincclark) [inline packages/pagination/pagination.js]: Can destruct `Component` at the top from react

File: packages/pagination/pagination.js [hunk 3]
```diff
@@ -40,67 +40,98 @@ const styles = StyleSheet.create({
...
+    );
+  }
+}
 
 Pagination.propTypes = {
   count: PropTypes.number,
   generatePageLink: PropTypes.func,
-  onNext: PropTypes.func,
-  onPrev: PropTypes.func,
+  onChangePage: PropTypes.func,
```
Comment (colincclark) [inline packages/pagination/pagination.js]: isn't this always a required prop?

Comment (colincclark) [reply packages/pagination/pagination.js]: if not, should have a story for pagination without pagination, lol

File: packages/pagination/__tests__/test-helper.js [hunk 4]
```diff
@@ -7,97 +7,75 @@ import React16Adapter from "enzyme-adapter-react-16";
 Enzyme.configure({ adapter: new React16Adapter() });
 
 export default Pagination => () => {
+  const checkMatchesSnapshot = props => {
```
Comment (colincclark) [inline packages/pagination/__tests__/test-helper.js]: 👍

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/5e8d7e65117c1702

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/43944a6b5e91b712

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/06de1481f82918ed

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/f9a93eb5e90e53c5

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/38bd244f8b7588cd

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/9b57ec74e54af13f

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/33b18c47b9b5fd23

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/c44043bb6ed1b8f9

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/e6ff7a12444755b8

**Comment by craigbilner** [comment]
closing in favour of debounce HoC

**Pull request closed without merging**